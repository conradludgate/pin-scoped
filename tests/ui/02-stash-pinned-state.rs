#![feature(async_closure)]

use std::{pin::pin, sync::OnceLock};

use pin_scoped::{Scope, ScopeState};

static STASH: OnceLock<ScopeState<'static, u64>> = OnceLock::new();

#[tokio::main]
async fn main() {
    let scoped = pin!(Scope::new(0));

    for _ in 0..64 {
        scoped.as_ref().spawn(async move |state: ScopeState<u64>| {
            STASH.set(state);
        });
    }

    scoped.await;
}
