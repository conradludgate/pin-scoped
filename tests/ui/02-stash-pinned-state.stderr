error[E0521]: borrowed data escapes outside of closure
  --> tests/ui/02-stash-pinned-state.rs:15:13
   |
14 |         scoped.as_ref().spawn(async move |state: ScopeState<u64>| {
   |                                           -----
   |                                           |
   |                                           `state` is a reference that is only valid in the closure body
   |                                           has type `ScopeState<'1, u64>`
15 |             STASH.set(state);
   |             ^^^^^^^^^^^^^^^^
   |             |
   |             `state` escapes the closure body here
   |             argument requires that `'1` must outlive `'static`
